apps/api/src/domain/models/user-management/user/commands/register-user.command.integration.spec.ts

```diff
@@ -0,0 +1,240 @@
+                    const userSearchResult = await testRepositoryProvider
+                        .getUserRepository()
+                        .fetchById(id);
+
+                    expect(userSearchResult).toBeInstanceOf(CoscradUser);
+                },
+            });
+        });
+    });
+
+    describe('when the command is invalid', () => {
+        describe('when the external state is invalid', () => {
+            describe('when there is already a user with the given id', () => {
+                it('should fail', async () => {
+                    const newId = await idManager.generate();
+
+                    await testRepositoryProvider.addFullSnapshot(
+                        buildInMemorySnapshot({ users: [existingUser.clone({ id: newId })] })
+                    );
+
+                    const validFSA = buildValidCommandFSA(newId);
+
+                    const executionResult =
+                        await commandAssertionDependencies.commandHandlerService.execute(validFSA);
+
+                    expect(executionResult).toBeInstanceOf(CommandExecutionError);
+
+                    // TODO Tighten this check up with custom error
+                    expect(
+                        (executionResult as CommandExecutionError).innerErrors[0]
+                    ).toBeInstanceOf(InternalError);
+                });
+            });
+
+            describe('when there is already a user with the given auth provider assigned user ID', () => {
+                it('should fail', async () => {
+                    const newId = await idManager.generate();
+
+                    await testRepositoryProvider.addFullSnapshot(
+                        buildInMemorySnapshot({ users: [existingUser] })
+                    );
+
+                    const invalidFSA = buildInvalidFSA(newId, {
+                        userIdFromAuthProvider: existingUser.authProviderUserId,
+                    });
+
+                    const executionResult =
+                        await commandAssertionDependencies.commandHandlerService.execute(
+                            invalidFSA
+                        );
+
+                    expect(executionResult).toBeInstanceOf(CommandExecutionError);
+
+                    // TODO Tighten this check up with custom error
+                    expect(
+                        (executionResult as CommandExecutionError).innerErrors[0]
+                    ).toBeInstanceOf(InternalError);
+                });
+            });
+
+            describe('when there is already a user with the given id', () => {
+                it('should fail', async () => {
+                    const newId = await idManager.generate();
+
+                    await testRepositoryProvider.addFullSnapshot(
+                        buildInMemorySnapshot({ users: [existingUser] })
+                    );
+
+                    const invalidFSA = buildInvalidFSA(newId, { username: existingUser.username });
```

Comment:
It seems a bit odd that we have 3 unique IDs for the user model, but I suppose this is standard. We should keep clear that the `id` is the source of truth (AggregateId` amongst these.